packages/react-scripts/scripts/init.js

```diff
@@ -17,10 +17,47 @@ process.on('unhandledRejection', err => {
 const fs = require('fs-extra');
 const path = require('path');
 const chalk = require('chalk');
+const execSync = require('child_process').execSync;
 const spawn = require('react-dev-utils/crossSpawn');
 const { defaultBrowsers } = require('react-dev-utils/browsersHelper');
 const os = require('os');
 
+function insideGitRepository() {
+  try {
+    execSync('git rev-parse --is-inside-work-tree', {stdio: 'ignore'});
+    return true;
+  } catch (e) {
+    return false;
+  }
+}
+
+function insideMercurialRepository() {
+  try {
+    execSync('hg --cwd . root', {stdio: 'ignore'});
+    return true;
+  } catch (e) {
+    return false;
+  }
+}
+
+function gitInit() {
+  try {
+    execSync('git --version', {stdio: 'ignore'});
+
+    if (insideGitRepository() || insideMercurialRepository()) {
+      return false;
+    }
+
+    execSync('git init', {stdio: 'ignore'});
+    execSync('git add .', {stdio: 'ignore'});
```

Comment:
Can this be `git add -A`?
(infact `git add .` covers create and update but this one feels more inclusive)